### ISSUE TITLE
SigmoidGrad legalization

### ISSUE THREAD

**Issue by bhack** [tensorflow/tensorflow#58645]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

master

### Custom Code

Yes

### Current Behaviour?

We had a legalization error with the MLIR bridge with the sigmoid in the focal loss:
https://github.com/keras-team/keras-cv/blob/master/keras_cv/losses/focal.py#L82

### Standalone code to reproduce the issue

https://colab.research.google.com/gist/ianstenbit/6cbcf6e034fbf63e22208f3d9a479297/sigmoid-loss-issue-retinanet-on-tpu.ipynb

See also:
https://github.com/keras-team/keras-cv/issues/1018

### Relevant log output

```python
(0) INVALID_ARGUMENT: {{function_node __inference_train_function_133447}} TF to XLA legalization failed: <unknown>:0: error: loc(fused["SigmoidGrad:", "gradient_tape/FocalLoss/Sigmoid/SigmoidGrad"]): 'mhlo.constant' op result #0 must be statically shaped tensor of floating-point or pred (AKA boolean or 1-bit integer) or 8/16/32/64-bit signless integer or 8/16/32/64-bit unsigned integer or complex type with 32-bit float or 64-bit float elements values, but got 'tensor<?x?x20xf32>'
<unknown>:0: note: loc(fused["SigmoidGrad:", "gradient_tape/FocalLoss/Sigmoid/SigmoidGrad"]): see current operation: %5996 = "mhlo.constant"() {value = dense<1.000000e+00> : tensor<?x?x20xf32>} : () -> tensor<?x?x20xf32>
```
</details>

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
